docs/code_gen_options.rst

```diff
@@ -218,4 +218,26 @@ always true, so the check is omitted.
 This also means that, whenever the length of an array is accessed using '.length', it is replaced with a constant.
 
 Note that this optimization does not cover every case. When an array is passed
-as a function argument, for instance, the length is unknown.
\ No newline at end of file
+as a function argument, for instance, the length is unknown.
```

Comment:
Please also change the first line and first paragraph of this file so it talks about code generation options, not just optimizations.